cmd/planner/main.go

```diff
@@ -73,18 +74,66 @@ func main() {
 		log.Fatalln("failed to connect to db:", err)
 	}
 
+	// Start AMQP consumer
 	consumer := queue.NewAmqpConsumer(rabbitMQUri, queueName, db, nil)
-	consumerCtx, cancel := context.WithCancel(ctx)
-	defer cancel()
+	var consumerWg sync.WaitGroup
+	consumerWg.Add(1)
 	go func() {
-		if err := consumer.Start(consumerCtx); err != nil {
-			log.Fatalf("failed to start AMQP consumer: %v", err)
+		defer consumerWg.Done()
+		if err := consumer.Start(ctx); err != nil && !errors.Is(err, context.Canceled) {
+			log.Printf("AMQP consumer error: %v", err)
```

Comment:
does this mean we won't exit the program even if there's error in starting the consumer?

Reply:
Oh thanks good catch!